crates/iroha_executor/src/permission.rs

```diff
@@ -647,6 +593,113 @@ pub mod asset_definition {
+    pub fn is_nft_owner(nft_id: &NftId, authority: &AccountId, host: &Iroha) -> Result<bool> {
+        let nft = host
+            .query(FindNfts)
+            .filter_with(|nft| nft.id.eq(nft_id.clone()))
+            .execute_single()
+            .map_err(|e| match e {
+                SingleQueryError::QueryError(e) => e,
+                SingleQueryError::ExpectedOneGotNone => {
+                    // assuming this can only happen due to such a NFT not existing
+                    ValidationFail::InstructionFailed(InstructionExecutionError::Find(
+                        FindError::Nft(nft_id.clone()),
+                    ))
+                }
+                _ => unreachable!(),
+            })?;
+        if nft.owned_by() == authority {
+            Ok(true)
+        } else {
+            domain::is_domain_owner(nft_id.domain(), authority, host)
+        }
+    }
+
+    /// Pass condition that checks if `authority` is the owner of NFT.
+    #[derive(Debug, Clone)]
+    pub struct Owner<'nft> {
+        /// NFT id to check against
+        pub nft: &'nft NftId,
+    }
+
+    impl PassCondition for Owner<'_> {
+        fn validate(&self, authority: &AccountId, host: &Iroha, _context: &Context) -> Result {
+            if is_nft_owner(self.nft, authority, host)? {
+                return Ok(());
+            }
+
+            Err(ValidationFail::NotPermitted(
+                "Can't access NFT owned by another account".to_owned(),
+            ))
+        }
+    }
+
+    impl ValidateGrantRevoke for CanRegisterNft {
+        fn validate_grant(&self, authority: &AccountId, context: &Context, host: &Iroha) -> Result {
+            super::domain::Owner::from(self).validate(authority, host, context)
+        }
+        fn validate_revoke(
+            &self,
+            authority: &AccountId,
+            context: &Context,
+            host: &Iroha,
+        ) -> Result {
+            super::domain::Owner::from(self).validate(authority, host, context)
+        }
+    }
+
+    macro_rules! impl_froms_and_validate_grant_revoke {
+        ($($name:ty),+ $(,)?) => {$(
+            impl<'t> From<&'t $name> for Owner<'t> {
+                fn from(value: &'t $name) -> Self {
+                    Self { nft: &value.nft }
+                }
+            }
+
+            impl ValidateGrantRevoke for $name {
+                fn validate_grant(&self, authority: &AccountId, context: &Context, host: &Iroha) -> Result {
+                    Owner::from(self).validate(authority, host, context)
+                }
+                fn validate_revoke(
+                    &self,
+                    authority: &AccountId,
+                    context: &Context,
+                    host: &Iroha,
+                ) -> Result {
+                    Owner::from(self).validate(authority, host, context)
+                }
+            }
+        )+};
+    }
+
+    impl_froms_and_validate_grant_revoke!(CanUnregisterNft, CanTransferNft, CanModifyNftMetadata);
```

Comment:
I don't think NFT contents are meant to be modified by owners after registration

Reply:
Looks like we should have the following rules, what do you think?
* Owner of `nft.domain` can unregister, transfer and modify NFT
* Owner of NFT can only transfer NFT

Reply:
It depends on the actual use case, but I think this is a reasonable default for now